### PR TITLE
[KeyVault] Make inherit better

### LINKED ISSUE #26390: Release stable version Azure.ResourceManager.KeyVault SDK

### PR DESCRIPTION
**Pull request by ArthurMa1978**

### CODE REVIEW

File: sdk/keyvault/Azure.ResourceManager.KeyVault/tests/Extensions.cs [hunk 1]
```diff
@@ -75,7 +75,7 @@ public static bool IsEqual(this DeletedVaultResource deletedVault, VaultData cre
             Assert.AreEqual(createdVault.Name, deletedVault.Data.Name);
             Assert.AreEqual(createdVault.Id, deletedVault.Data.Properties.VaultId);
             Assert.AreEqual("Microsoft.KeyVault/deletedVaults", deletedVault.Data.ResourceType);
-            Assert.True(createdVault.Tags.DictionaryEqual(deletedVault.Data.Properties.Tags));
+            Assert.True(createdVault.Tags.DictionaryEqual(deletedVault.Data.Properties.Tags.ToDictionary(kvp => kvp.Key, kvp => kvp.Value)));
```
Comment (ArcturusZhang) [inline sdk/keyvault/Azure.ResourceManager.KeyVault/tests/Extensions.cs:78]: why do we need this?

File: sdk/keyvault/Azure.ResourceManager.KeyVault/src/autorest.md [hunk 2]
```diff
@@ -90,6 +90,7 @@ directive:
       $.VaultCreateOrUpdateParameters.properties.location['x-ms-format'] = 'azure-location';
       $.VaultAccessPolicyParameters.properties.location['x-ms-format'] = 'azure-location';
       $.Vault.properties.location['x-ms-format'] = 'azure-location';
+      $.Vault['x-csharp-usage'] = 'model,input,output';
```
Comment (fengzhou-msft) [inline sdk/keyvault/Azure.ResourceManager.KeyVault/src/autorest.md:93]: Shall we try ` $.Vault['x-csharp-usage'] = 'model,output';` as VaultData is not used in requests?

File: sdk/keyvault/Azure.ResourceManager.KeyVault/tests/Extensions.cs (hunk 1 above)
Comment (ArthurMa1978) [reply sdk/keyvault/Azure.ResourceManager.KeyVault/tests/Extensions.cs:78]: deletedVault.Data.Properties.Tags is `IReadOnlyDictionary` but `createdVault.Tags` is IDictionary`

File: sdk/keyvault/Azure.ResourceManager.KeyVault/src/autorest.md (hunk 2 above)
Comment (ArthurMa1978) [reply sdk/keyvault/Azure.ResourceManager.KeyVault/src/autorest.md:93]: Add `input` is used to make it's properties type match with `TrackedResourceData`

**Review by fengzhou-msft — APPROVED** [review]

**Pull request merged**